code/modules/projectiles/guns/energy/stun.dm

```diff
@@ -30,7 +31,51 @@
 	origin_tech = list(TECH_COMBAT = 3, TECH_MATERIAL = 3, TECH_POWER = 2)
 	projectile_type = /obj/item/projectile/energy/electrode
 	max_shots = 8
+	var/emagged = FALSE
 
+/obj/item/gun/energy/stunrevolver/verb/rename_gun()
+	set name = "Name Gun"
+	set category = "Object"
+	set desc = "Click to rename your gun. If you're the detective."
+
+	var/mob/M = usr
+	if(!M.mind)
+		return FALSE
+	if(name != initial(name))
+		to_chat(M, SPAN_WARNING("\The [src] has already been renamed!"))
+		return
+
+	var/input = sanitizeSafe(input(M, "What do you want to name the gun?", "Gun Naming"), MAX_NAME_LEN)
+
+	if(src && input && !M.stat && in_range(M, src))
+		name = input
+		to_chat(M, SPAN_NOTICE("You name the gun <b>[input]</b>. Say hello to your new friend."))
+		return TRUE
+
+/obj/item/gun/energy/stunrevolver/verb/spin_cylinder()
+	set name = "Spin Cylinder"
+	set desc = "Fun when you're bored out of your skull."
+	set category = "Object"
+
+	usr.visible_message(SPAN_WARNING("\The [usr] spins the cylinder of \the [src]!"), SPAN_WARNING("You spin the cylinder of \the [src]!"), SPAN_NOTICE("You hear something metallic spin and click."))
+	playsound(get_turf(src), 'sound/weapons/revolver_spin.ogg', 100, TRUE)
+
+/obj/item/gun/energy/stunrevolver/emag_act(remaining_charges, mob/user)
+	if(!remaining_charges)
+		return FALSE
+	if(emagged)
+		to_chat(user, SPAN_WARNING("\The [src] has already been emagged!"))
+		return
+	to_chat(user, SPAN_NOTICE("You short out \the [src]'s limiter, supercharging its power cylinder."))
+	projectile_type = /obj/item/projectile/energy/blaster
+	fire_sound = 'sound/weapons/laserstrong.ogg'
+	power_supply.maxcharge = 10000
+	power_supply.charge = 10000
+	charge_cost = 1000
+	self_recharge = TRUE
+	recharge_time = 5
```

Comment:
That there is a nogo (even if its just a antag-only thing).
I do not like the idea of "emag your standard sidearm and you have a weapon that can fire a unlimited number of lethal shots"

Reply:
I think the antagonist spin on this is super good. Why would it be bad to have?